### ISSUE TITLE
OkHttp ConnectionPool leaks

### ISSUE THREAD

**Issue by lordpengwin** [java-operator-sdk/java-operator-sdk#625]
## Bug Report

I'm using version 1.9.4 of the java-operator-sdk with corresponding version of the quarkus-operator-sdk and I'm constantly getting the following warnings:

```
2021-10-26 13:41:25,656 WARNING [okh.OkHttpClient] (OkHttp ConnectionPool) A connection to https://10.96.0.1/ was leaked. Did you forget to close a response body?: java.lang.Throwable: response.body().close()
	at okhttp3.internal.platform.Platform.getStackTraceForCloseable(Platform.java:149)
	at okhttp3.internal.connection.Transmitter.callStart(Transmitter.java:116)
	at okhttp3.RealCall.execute(RealCall.java:78)
	at io.fabric8.kubernetes.client.dsl.internal.ClusterOperationsImpl.handleVersionGet(ClusterOperationsImpl.java:62)
	at io.fabric8.openshift.client.internal.OpenShiftClusterOperationsImpl.fetchOpenshift4Version(OpenShiftClusterOperationsImpl.java:56)
	at io.fabric8.openshift.client.internal.OpenShiftClusterOperationsImpl.fetchVersion(OpenShiftClusterOperationsImpl.java:48)
	at io.fabric8.openshift.client.DefaultOpenShiftClient.getVersion(DefaultOpenShiftClient.java:568)
	at io.javaoperatorsdk.operator.Operator.start(Operator.java:68)
	at com.xxx.kubernetes.operators.AppserverOperator.run(AppserverOperator.java:45)
	at com.xxx.kubernetes.operators.AppserverOperator_ClientProxy.run(AppserverOperator_ClientProxy.zig:129)
	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:123)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:66)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:42)
	at com.kasisto.kubernetes.operators.AppserverOperator.main(AppserverOperator.java:37)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:48)
	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:25)
```
 I'm using the fabric8 kubernetes client (5.7.3 i think) to create and manage custom resources, so i don't really have access to the underlying requests so I'm not sure what code would be responsible for closing the response body.

#### What did you do?

Create a custom resource and then I get this warning when it is deleted.

#### What did you expect to see?

No warnings or errors

#### What did you see instead? Under which circumstances?

This happens all that time, generally after a few minutes of inactivitiy between creating the resource and deleting it.

#### Environment

- Mac using the local docker desktop kebernetes
- Quarkus 2.3.0-Final
- java-operator-framework and quarkus-operator-sdk 1.9.4
- fabric8 5.7.3

`$ 1.9.4`

`$ java version "11.0.9" 2020-10-20 LTS`

`$ Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.5", GitCommit:"aea7bbadd2fc0cd689de94a54e5b7b758869d691", GitTreeState:"clean", BuildDate:"2021-09-15T21:10:45Z", GoVersion:"go1.16.8", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.5", GitCommit:"aea7bbadd2fc0cd689de94a54e5b7b758869d691", GitTreeState:"clean", BuildDate:"2021-09-15T21:04:16Z", GoVersion:"go1.16.8", Compiler:"gc", Platform:"linux/amd64"}`

**Comment by sclorng** [comment]
@metacosm should be closed with 1.10.2

**Comment by metacosm** [comment]
> @metacosm should be closed with 1.10.2

Indeed.
